### DISCUSSION TITLE
Middleware between monaco-editor and LSP

### DISCUSSION THREAD

**Discussion by ghost** [eclipse-langium/langium#1419] [Q&A]
Can middleware be added between editors monaco-editor and LSP to intercept communication between them? Release after processing, for example, I want to add some header files, but do not want to show this header file to the user, so add this header file when monaco-editor communicates with LSP. Delete the header file when LSP communicates with monaco-editor.

**Comment by msujew** [comment]
Hey @jlxzq,

this isn't really related to Langium - we neither build the monaco editor nor the language client packages here. But you should be able to register [any middleware on the `LanguageClient`](https://github.com/microsoft/vscode-languageserver-node/blob/ed3cd0f78c1495913bda7318ace2be7f968008af/client/src/common/client.ts#L356) from the `vscode-languageserver-client` npm package. For any further questions you might want to ask the people over at the [`vscode-languageserver-node`](https://github.com/microsoft/vscode-languageserver-node) repo or at the [`monaco-languageclient`](https://github.com/TypeFox/monaco-languageclient/).
